src/services/hsm.service.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const { hsm } = require('@dinamonetworks/hsm-dinamo');
+
+const { getConfig } = require('../settings');
+const { lock, lockTypes } = require('../lock');
+const { hsmBusyErrorMessage } = require('../helpers/constants');
+const { HsmError } = require('../errors');
+
+/**
+ * Connects to the HSM and returns the connection object
+ * @returns {Promise<Hsm>}
+ */
+async function hsmConnect() {
+  // Ensures that only one connection will be open at any given moment
```

Comment:
Can't you reuse the same connection to start other wallets?

Reply:
While using the dinamo lib we currently can't. [A comment](https://github.com/HathorNetwork/rfcs/pull/67#discussion_r1462591197) was made on the design about this specific point.